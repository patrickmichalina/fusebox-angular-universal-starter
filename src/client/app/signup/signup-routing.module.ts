import { SignupComponent } from './signup.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { MetaGuard } from '@ngx-meta/core'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: SignupComponent,
        canActivate: [MetaGuard],
        data: {
          meta: {
            title: 'i18n.signup.title',
            description: 'i18n.signup.description'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class SignupRoutingModule { }
