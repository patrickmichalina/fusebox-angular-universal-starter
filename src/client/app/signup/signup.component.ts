import { ChangeDetectionStrategy, Component } from '@angular/core'

@Component({
  selector: 'pm-signup',
  templateUrl: './signup.component.html',
  styleUrls: ['./signup.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SignupComponent { }
