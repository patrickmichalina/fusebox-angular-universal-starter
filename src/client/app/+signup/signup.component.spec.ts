import { SignupComponent } from './signup.component';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { SignupModule } from './signup.module';

describe(SignupComponent.name, () => {
  let fixture: ComponentFixture<SignupComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SignupModule],
      declarations: [TestComponent]
    }).compileComponents();
  }));

  beforeEach(async(() => {
    fixture = TestBed.createComponent(SignupComponent);
    fixture.detectChanges();
  }));

  afterEach(async(() => {
    TestBed.resetTestingModule();
  }));

  it('should match snapshot', async(() => {
    expect(fixture).toMatchSnapshot();
  }));

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined();
  }));
});

@Component({
  selector: 'test-component',
  template: '<pm-signup></pm-signup>'
})
class TestComponent {}
