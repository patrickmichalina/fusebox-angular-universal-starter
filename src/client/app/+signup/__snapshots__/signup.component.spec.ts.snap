// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupComponent should match snapshot 1`] = `
<pm-signup>
  Your <strong>
    signup
  </strong> form goes here!
</pm-signup>
`;
